### PR TITLE
improve reflection helper.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @vicancy @hellosnow @ansyral @qinezh @superyyrrzz

### CODE REVIEW

**Comment by vicancy** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/ReflectionHelper.cs [hunk 1]
```diff
@@ -36,89 +40,100 @@ public static List<PropertyInfo> GetGettableProperties(Type type)
...
-            foreach (var i in type.GetInterfaces())
-            {
-                if (IsGenericType(i, genericTypeDefinition))
-                {
-                    return true;
-                }
-            }
-
-            return false;
+            return GetGenericType(type, genericTypeDefinition) != null;
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/ReflectionHelper.cs]: >GetGenericType [](start = 19, length = 14)

return TryGetGenericType?

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/ReflectionHelper.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Common/ModelAttributeHandlers/ReflectionHelper.cs]: I want to remove `TryGetGenericType`.

---
In reply to: [96348446](https://github.com/dotnet/docfx/pull/1187#discussion_r96348446) [](ancestors = 96348446)

**Pull request merged**